src/commands/synthetics/__tests__/cli.test.ts

```diff
@@ -395,44 +451,473 @@ describe('run-test', () => {
+          `body=${defaultTestOverrides.body}`,
+          `bodyType=${defaultTestOverrides.bodyType}`,
+          `cookies=${(defaultTestOverrides.cookies as Cookies).value}`,
+          `cookies.append=${(defaultTestOverrides.cookies as Cookies).append}`,
+          `defaultStepTimeout=${defaultTestOverrides.defaultStepTimeout}`,
+          `deviceIds=${defaultTestOverrides.deviceIds?.join(';')}`,
+          `executionRule=${defaultTestOverrides.executionRule}`,
+          `followRedirects=${defaultTestOverrides.followRedirects}`,
+          `headers.Content-Type=${defaultTestOverrides.headers ? defaultTestOverrides.headers['Content-Type'] : ''}`,
+          `headers.Authorization=${defaultTestOverrides.headers?.Authorization}`,
+          `locations=${defaultTestOverrides.locations?.join(';')}`,
+          `retry.count=${defaultTestOverrides.retry?.count}`,
+          `retry.interval=${defaultTestOverrides.retry?.interval}`,
+          `startUrl=${defaultTestOverrides.startUrl}`,
+          `startUrlSubstitutionRegex=${defaultTestOverrides.startUrlSubstitutionRegex}`,
+          `testTimeout=${defaultTestOverrides.testTimeout}`,
+          `resourceUrlSubstitutionRegexes=${defaultTestOverrides.resourceUrlSubstitutionRegexes?.join(';')}`,
+          `variables.cliVar1=${defaultTestOverrides.variables?.cliVar1}`,
+          `variables.cliVar2=${defaultTestOverrides.variables?.cliVar2}`,
+        ]
 
-      await command['resolveConfig']()
-      expect(command['config']).toEqual({
-        ...DEFAULT_COMMAND_CONFIG,
-        apiKey: 'api_key_cli',
-        appKey: 'app_key_env',
-        datadogSite: 'us5.datadoghq.com',
-        global: {
-          pollingTimeout: 111,
-          mobileApplicationVersionFilePath: './path/to/application_config_file.apk',
-        },
-        defaultTestOverrides: {
-          pollingTimeout: 333,
-          mobileApplicationVersionFilePath: './path/to/application_cli.apk',
-        },
+        await command['resolveConfig']()
+
+        // TODO SYNTH-12989: Clean up deprecated `global`, `location`, `variableStrings`, `mobileApplicationVersionFilePath`, `proxy` etc.
+        // This fixes are only here for the test to run, and to maintain backward compatibility.
+        const {mobileApplicationVersionFilePath, pollingTimeout, ...filteredOverrideCLI} = overrideCLI
+        const expectedCLIOverrideResult = {
+          ...filteredOverrideCLI,
+          locations: configFile.locations,
+          global: {},
+          defaultTestOverrides: {
+            ...defaultTestOverrides,
+            mobileApplicationVersionFilePath,
+            pollingTimeout,
+          },
+          pollingTimeout: configFile.pollingTimeout,
+          proxy: configFile.proxy,
+          variableStrings: configFile.variableStrings,
+        }
+
+        expect(command['config']).toEqual(expectedCLIOverrideResult)
+      })
+      test('ENV < CLI', async () => {
```

Comment:
For this test you have some values in the `overrideEnv` that are the same as the `cofigFile`. I've pointed them out in case you want to chnage them, but since you're testing `env < cli` not sure if that matters as much. As far as i can see the values for env and cli are different 👍

Reply:
I think we can ignore it as the `configFile` is not being used in this test 😄